osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineHitObjectBlueprint.cs

```diff
@@ -117,12 +146,54 @@ public TimelineHitObjectBlueprint(HitObject hitObject)
             }
             else
             {
-                AddInternal(circle);
+                mainComponents.Add(circle);
             }
 
             updateShadows();
         }
 
+        [BackgroundDependencyLoader(true)]
+        private void load(HitObjectComposer composer)
+        {
+            if (composer != null)
+            {
+                // best effort to get the drawable representation for grabbing colour and what not.
+                drawableHitObject = composer.HitObjects.FirstOrDefault(d => d.HitObject == HitObject);
+            }
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            if (HitObject is IHasComboInformation comboInfo)
+            {
+                comboIndex = comboInfo.IndexInCurrentComboBindable.GetBoundCopy();
+                comboIndex.BindValueChanged(combo =>
+                {
+                    comboIndexText.Text = (combo.NewValue + 1).ToString();
+                }, true);
+            }
+
+            if (drawableHitObject != null)
+            {
+                comboColour = drawableHitObject.AccentColour.GetBoundCopy();
+                comboColour.BindValueChanged(colour =>
+                {
+                    if (HitObject is IHasDuration)
+                        mainComponents.Colour = ColourInfo.GradientHorizontal(drawableHitObject.AccentColour.Value, Color4.White);
+                    else
+                        mainComponents.Colour = drawableHitObject.AccentColour.Value;
+
+                    var col = mainComponents.Colour.AverageColour.Linear;
+                    float brightness = col.R + col.G + col.B;
```

Comment:
Not sure this should use `AverageColour`. Due to averaging over the gradient it can lead to some funky-looking results in some edge cases (hitobjects with duration and dark combo colour):

![2020-09-29-223705_693x95_scrot](https://user-images.githubusercontent.com/20418176/94613153-6d811400-02a4-11eb-9ed6-aea1551a0fc0.png)

(this screenshot was taken on a dirty tree with `circle_size == 32`)

About the only place where I can think always using `AccentColour` could be a problem is on extreme zoom-out.

Reply:
Oh good point. I was using average before I added the gradient..